### ISSUE TITLE
Error when using `TypeVar` defaults and `Self`

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#4418]
**Describe the bug**
when using a TypeVar with generics in a function inside a class along with the method returning `Self` pyright gives an error which it should not.

**To Reproduce**
Create a generic class with a `TypeVar` which has a default and attempt to make a method which takes `T` and returns `Self`

**Expected behavior**
It should not error as it is a false error.

**Screenshots or Code**
![Code_-_Insiders_Gh4nJH7mdW](https://user-images.githubusercontent.com/39768508/211158006-6d9da454-7a96-41a1-9539-826abe0b2da0.png)

```py
T = TypeVar("T", default=int)


class Foo(Generic[T]):
    def __init__(self, value: T):
        self.value = value

    def update(self, value: T) -> Self:
        self.value = value
        return self
```

```
  /home/zoma/pyright_bug.py:13:9 - error: "__type_of_self__" cannot appear after "T" in type parameter list because it has no default type (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
`pyright 1.1.287`

**Comment by erictraut** [comment]
Thanks for the bug report. I'm glad that someone is trying out the PEP 696 functionality.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.288, which I just published. It will also be included in a future release of pylance.
